src/lib.rs

```diff
@@ -53,7 +56,7 @@ pub struct AstarteSdk {
     mqtt_options: MqttOptions,
     client: AsyncClient,
     eventloop: Arc<tokio::sync::Mutex<EventLoop>>,
-    interfaces: interfaces::Interfaces,
+    interfaces: Arc<tokio::sync::Mutex<interfaces::Interfaces>>,
```

Comment:
Why are we using `tokio::sync::Mutex` here? [This section](https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use) seems to imply that we would be better off using the one in the standard library or in `parking_lot` (that we already have as indirect dependency).
Moreover, can't we use a `RwLock` instead of a Mutex? I don't think we need to lock it every time we read from it, only when we write to it